### PR TITLE
Fix updateStats test failing sometimes on Appveyor

### PR DESCRIPTION
**Pull request by gumb0**
- make updateStats test use NoProof like other unit tests instead of full Ethash
- instead of hard-coded values, calculate expected values based on data structure sizes

### CODE REVIEW

File: test/unittests/libethereum/BlockChain.cpp [hunk 1]
```diff
@@ -379,13 +374,25 @@ BOOST_AUTO_TEST_CASE(updateStats)
...
-    BOOST_REQUIRE_EQUAL(stat.memBlockHashes, 0);
-    BOOST_REQUIRE_EQUAL(stat.memBlocks, 675);
-    BOOST_REQUIRE_EQUAL(stat.memDetails, 138);
-    BOOST_REQUIRE_EQUAL(stat.memLogBlooms, 8422);
-    BOOST_REQUIRE_EQUAL(stat.memReceipts, 0);
-    BOOST_REQUIRE_EQUAL(stat.memTotal(), 9235);
-    BOOST_REQUIRE_EQUAL(stat.memTransactionAddresses, 0);
+    BOOST_CHECK_EQUAL(stat.memBlockHashes, 0);
+
+    unsigned const memBlocksExpected = block.bytes().size() + 64;
```
Comment (gumb0) [inline test/unittests/libethereum/BlockChain.cpp]: Not sure why the implementation adds this `+ 64`. Pointer size? Two hashes?

**Comment by gumb0** [comment]
Now another test hang up 😕

**Comment by winsvega** [comment]
which test hang?

File: test/unittests/libethereum/BlockChain.cpp [hunk 2]
```diff
@@ -353,24 +353,19 @@ BOOST_AUTO_TEST_CASE(rescue, *utf::expected_failures(1))
...
-
-BOOST_FIXTURE_TEST_SUITE(BlockChainSuite, TestOutputHelperFixture)
-
 BOOST_AUTO_TEST_CASE(updateStats)
 {
     TestBlockChain bc(TestBlockChain::defaultGenesisBlock());
     BlockChain& bcRef = bc.interfaceUnsafe();
 
     BlockChain::Statistics stat = bcRef.usage();
-    //Absolutely random values here!
```
Comment (chfast) [inline test/unittests/libethereum/BlockChain.cpp]: LOL.

**Comment by gumb0** [comment]
@winsvega it's not a deadlock, it's PoW mining sometimes taking too much time.
Ethash should have it's own unit tests separate from testing Blockchain or BlockQueue. And it has some.

After I changed `updateTests` another one that hanged was `BlockQueueImport`, so I changed it to use `NoProof`, too

**Comment by winsvega** [comment]
Could it be because of the DAG generation?

**Pull request merged**